content-api-post.js

```diff
@@ -36,129 +38,129 @@ function metadataEscape(input) {
           }
 
-          const s3Params = {
-            ACL: 'public-read',
-            Body: processedImage,
-            Bucket: process.env.S3_BUCKET,
-            ContentType: mimeType.mime,
-            Key: path,
-            Metadata: metadata,
-          };
-
-          const s3 = new AWS.S3({signatureVersion: 'v4'});
-          s3.putObject(s3Params, function (error) {
-            if (error) {
-              return reject(fail(`Save error: ${error}. Metadata: ` + JSON.stringify(metadata), 500));
-            }
-
-            console.log('Successfully uploaded to S3, path: ' + path);
-            return resolve({
-              statusCode: 200,
-              body: JSON.stringify({
-                'uri': `${process.env.IMAGE_ACCESS_BASE_URI}/${path}`,
-              }),
-              isBase64Encoded: false,
-              headers: {
-                'Content-Type': 'application/json',
-              },
-            });
+          console.log('Successfully uploaded to S3, path: ' + path);
+          return resolve({
+            statusCode: 200,
+            body: JSON.stringify({
+              'uri': `${process.env.IMAGE_ACCESS_BASE_URI}/${path}`,
+            }),
+            isBase64Encoded: false,
+            headers: {
+              'Content-Type': 'application/json',
+            },
           });
-        })
-
-        .catch(sharpError => {
-          /**
-           * @param {Error} sharpError
-           */
-          if (sharpError.message.includes('VipsJpeg: Invalid SOS parameters for sequential JPEG')) {
-            return reject(fail('Processing error: corrupt JPEG, invalid SOS parameters', 400));
-          }
-
-          if (sharpError.message.includes('VipsJpeg: Premature end of input file')) {
-            return reject(fail('Processing error: corrupt JPEG, premature end of input file', 400));
-          }
-
-          if (sharpError.message.includes('Input buffer contains unsupported image format')) {
-            return reject(fail('Processing error: unsupported image format', 400));
-          }
-
-          return reject(fail(`Processing error: ${sharpError.message}`, 500));
-        })
+        });
+      })
+
+      .catch(sharpError => {
+        /**
+         * @param {Error} sharpError
+         */
+        if (sharpError.message.includes('VipsJpeg: Invalid SOS parameters for sequential JPEG')) {
+          return resolve(fail('Processing error: corrupt JPEG, invalid SOS parameters', 400));
+        }
+
+        if (sharpError.message.includes('VipsJpeg: Premature end of input file')) {
+          return resolve(fail('Processing error: corrupt, premature end of input file', 400));
+        }
+
+        if (sharpError.message.includes('Input buffer contains unsupported image format')) {
+          return resolve(fail('Processing error: unsupported image format', 400));
+        }
+
+        return resolve(fail(`Processing error: ${sharpError.message}`, 500));
```

Comment:
in this one case should we not reject the promise as this is a 500 error - I guess this line should never be hit but will be if we miss any possible reason for error in the previous series of if statements.

Reply:
@bdsl IMO, no, it is still better to return a 500 to the client rather than having API Gateway treat it as a crash and having it return a generic 502 without our message.

Reply:
@NoelLH Yes that does make sense, thanks. I see if we reject then the gateway always treats that as an internal issue that should be hidden from the client, but this message is better to share. Will approve PR.